src/App.jsx

```diff
@@ -1,35 +1,49 @@
-import { useState } from 'react'
-import reactLogo from './assets/react.svg'
-import viteLogo from '/vite.svg'
-import './App.css'
+import Home from "./pages/Home";
+import Articles from "./pages/Articles"
+import Header from "./components/Header";
+import NavBar from "./components/NavBar";
+
+import { getArticles } from "./utils/api";
+
+import { useState, useEffect } from "react";
+import { Routes, Route } from "react-router-dom";
+
+import "./App.css";
 
 function App() {
-  const [count, setCount] = useState(0)
+  const [articles, setArticles] = useState([]);
+  const [error, setError]= useState(null)
+  const [isLoading, setIsLoading] = useState(true)
+
+  useEffect(()=>{
+    setError(null)
+    getArticles()
+    .then((articles)=>{
+      setArticles(articles.data.articles)
+      setIsLoading(false)
+    })
+    .catch((error) => {
+      setError("Error fetching data")
+      setIsLoading(false)
+    })
+  }, [])
+
+  if (isLoading) {
+    return <p>Loading...</p>}
+  if (error) {
+    return <p>{error}</p>
```

Comment:
Is the actual error message you really what you want to show the user? Will it make sense to them? Probably not.

Think about a more user friendly error message to present.